kotlinpoet/src/main/java/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -626,6 +626,9 @@ internal class CodeWriter constructor(
    *
    * If [modifiers] contains [KModifier.PUBLIC], this method always returns `true`.
    *
+   * Next, if [implicitModifiers] contains [KModifier.PUBLIC] and [modifiers] contains
```

Comment:
Not a fan of comments that restate what's happening in the code, but this comment has already been like that before your changes. I'd suggest reverting this change, and we can clean up (or delete) the comment in a separate PR.

Reply:
Ok, I will remove this comment.